platform/src/lib.rs

```diff
@@ -545,6 +559,66 @@ pub extern "C" fn roc_fx_fileReadBytes(
     }
 }
 
+#[no_mangle]
+pub extern "C" fn roc_fx_fileReader(roc_path: &RocList<u8>) -> RocResult<u64, roc_app::ReadErr> {
+    match File::open(path_from_roc_path(roc_path)) {
+        Ok(file) => READERS.with(|reader_thread_local| {
+            let mut readers = reader_thread_local.borrow_mut();
+
+            let new_index = readers.len();
+
+            let buf_reader = BufReader::new(file);
+
+            readers.push(Rc::new(RefCell::new(Box::new(buf_reader))));
+
+            RocResult::ok(new_index as u64)
+        }),
+        Err(err) => RocResult::err(toRocReadError(err)),
+    }
+}
+
+#[no_mangle]
+pub extern "C" fn roc_fx_fileReadLine(readerIndex: u64) -> RocResult<RocList<u8>, RocStr> {
+    READERS.with(|reader_thread_local| {
+        let readers = reader_thread_local.borrow_mut();
+
+        let reader_ref_cell = readers.get(readerIndex as usize).expect("readerIndex was larger than number of readers in Vec. We expect this can only happen due to an implementation error in basic-cli.");
+
+        let mut string_buffer = String::new();
+
+        let mut file_buf_reader = reader_ref_cell.borrow_mut();
+
+        match file_buf_reader.read_line(&mut string_buffer) {
+            Ok(bytes_read) => {
+                if bytes_read == 0 {
```

Comment:
It seems like even without this clause the `string_buffer` would be empty, so it's probably unnecessary. I could understand keeping it for readability reasons.

Reply:
Ah, thank you.